pkg/apis/deployment/v1alpha/bootstrap.go

```diff
@@ -78,13 +98,13 @@ func (b *BootstrapSpec) Validate() error {
 // SetDefaults fills in default values when a field is not specified.
 func (b *BootstrapSpec) SetDefaults(deploymentname string) {
```

Comment:
This function takes me long to understand.
You could simply by having these 3 steps:
- check if map is nil -> make new (empty) map
- check if map contains root, if not add it with Auto value
- foreach entry in map, if Auto -> replace with calculated secret name